UiPath.FreeRdpClient/UiPath.FreeRdpClient/Internals/FreeRdpClient.cs

```diff
@@ -62,7 +61,10 @@ public async Task<IAsyncDisposable> Connect(RdpConnectionSettings connectionSett
 
         Task<AsyncDisposable> DoConnect() => Task.Run(() =>
         {
-            NativeInterface.RdpLogon(connectOptions, out var releaseObjectName);
+            NativeInterface.RdpLogon(
+                rdpOptions: connectOptions,
+                freeRdpDisconnectedCallback: () => connectionSettings.DisconnectCallback?.Invoke(),
+                out var releaseObjectName);
```

Comment:
the callback in this case becomes a candidate for GC after RdpLogon
so you need to save it and store it for the lifetime of the returned asyncDisposable